### ISSUE TITLE
merge_asof(..., left_on='a', right_index=True)

### ISSUE THREAD

**Issue by noreentry** [dask/dask#6183]
Hello!

There is an issue with **dask.dataframe.merge_asof()** when merging column in the left df with index in the right:

```python
nparts = 1
dd.merge_asof(dd.from_pandas(pd.DataFrame(index=[10,20,30], data={'a':[5, 15,25]}), npartitions=nparts),
              dd.from_pandas(pd.DataFrame(index=[10,20,30], data={'b':[10,20,30]}), npartitions=nparts),
              left_on='a', right_index=True).compute()
```

  | a | b
-- | -- | --
0 | 5 | NaN
1 | 15 | 10.0
2 | 25 | 20.0
3 | 5 | NaN
4 | 15 | 10.0
5 | 25 | 20.0

Splitting data into 3 partitions i get:
```python
nparts = 3
dd.merge_asof(dd.from_pandas(pd.DataFrame(index=[10,20,30], data={'a':[5, 15,25]}), npartitions=nparts),
              dd.from_pandas(pd.DataFrame(index=[10,20,30], data={'b':[10,20,30]}), npartitions=nparts),
              left_on='a', right_index=True).compute()
```

  | a | b
-- | -- | --
0 | 5 | NaN
1 | 5 | NaN
0 | 15 | 10.0
1 | 25 | 10.0
2 | 15 | 10.0
3 | 25 | 20.0

Expected result is:
```python
pd.merge_asof(pd.DataFrame(index=[10,20,30], data={'a':[5, 15,25]}),
              pd.DataFrame(index=[10,20,30], data={'b':[10,20,30]}),
              left_on='a', right_index=True)
```

  | a | b
-- | -- | --
10 | 5 | NaN
20 | 15 | 10.0
30 | 25 | 20.0

**Comment by jcrist** [comment]
Thanks for the excellent reproducible issue @noreentry. This indeed is a bug. Would you be interested in looking deeper into this issue?

**Comment by jcrist** [comment]
Fixed by #6192.
